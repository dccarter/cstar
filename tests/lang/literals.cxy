/*
 * Tests literals in cxy
 */

func main(args: [string]) {
    // null literal
    null;
    // bool literal
    true;
    false;
    // char literals
    'a'; 'b'; '\n'; '\0'; 'ç•Œ'; 'ðŸ˜Š';
    // integers
    0; 100; 18446744073709551615;
    -1; -2; -9223372036854775807;
    0b100; 0b1010;
    0o777; 0o189;
    0xa; 0xff; 0xffffff;
    // floats
    1.100; 1.000; 10.0e5; 1.9e-5;
    // strings
    ""; "Hello World"; "Hello ä¸–ç•Œ";
}
