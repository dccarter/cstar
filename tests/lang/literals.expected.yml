Program:
decls:
- FuncDecl:
  flags: (TopLevelDecl)
  name: main
  params:
  - FuncParam:
    name: args
    index: 0
    paramType:
      ArrayType:
      flags: (TypeAst)
      element:
        StringType:
        flags: (TypeAst)
  body:
    BlockStmt:
    stmts:
    - ExprStmt:
      expr:
        NullLit:
        value: null
    - ExprStmt:
      expr:
        BoolLit:
        value: true
    - ExprStmt:
      expr:
        BoolLit:
        value: false
    - ExprStmt:
      expr:
        CharLit:
        value: 'a'
    - ExprStmt:
      expr:
        CharLit:
        value: 'b'
    - ExprStmt:
      expr:
        CharLit:
        value: '

          '
    - ExprStmt:
      expr:
        CharLit:
        value: ''
    - ExprStmt:
      expr:
        CharLit:
        value: "\u754C"
    - ExprStmt:
      expr:
        CharLit:
        value: "\U0001F60A"
    - ExprStmt:
      expr:
        IntegerLit:
        value: 0
    - ExprStmt:
      expr:
        IntegerLit:
        value: 100
    - ExprStmt:
      expr:
        IntegerLit:
        value: -1
    - ExprStmt:
      expr:
        UnaryExpr:
        op: 24
        isPrefix: true
        lhs:
          IntegerLit:
          value: 1
    - ExprStmt:
      expr:
        UnaryExpr:
        op: 24
        isPrefix: true
        lhs:
          IntegerLit:
          value: 2
    - ExprStmt:
      expr:
        UnaryExpr:
        op: 24
        isPrefix: true
        lhs:
          IntegerLit:
          value: 9223372036854775807
    - ExprStmt:
      expr:
        IntegerLit:
        value: 4
    - ExprStmt:
      expr:
        IntegerLit:
        value: 10
    - ExprStmt:
      expr:
        IntegerLit:
        value: 511
    - ExprStmt:
      expr:
        IntegerLit:
        value: 1
    - ExprStmt:
      expr:
        IntegerLit:
        value: 89
    - ExprStmt:
      expr:
        IntegerLit:
        value: 10
    - ExprStmt:
      expr:
        IntegerLit:
        value: 255
    - ExprStmt:
      expr:
        IntegerLit:
        value: 16777215
    - ExprStmt:
      expr:
        FloatLit:
        value: 1.100000
    - ExprStmt:
      expr:
        FloatLit:
        value: 1.000000
    - ExprStmt:
      expr:
        FloatLit:
        value: 1000000.000000
    - ExprStmt:
      expr:
        FloatLit:
        value: 0.000019
    - ExprStmt:
      expr:
        StringLit:
        value: ""
    - ExprStmt:
      expr:
        StringLit:
        value: "Hello World"
    - ExprStmt:
      expr:
        StringLit:
        value: "Hello \u4E16\u754C"
